includes/Modules/Sign_In_With_Google.php

```diff
@@ -152,9 +155,18 @@ public function handle_login_errors( $error ) {
 	/**
 	 * Cleans up when the module is deactivated.
 	 *
+	 * Persist the clientID on module disconnection, so it can be
+	 * reused if the module were to be reconnected.
+	 *
 	 * @since 1.137.0
 	 */
 	public function on_deactivation() {
+		$pre_deactivation_settings = $this->get_settings()->get();
+
+		if ( isset( $pre_deactivation_settings['clientID'] ) ) {
+			( new Existing_Client_ID( $this->options ) )->set( $pre_deactivation_settings['clientID'] );
+		}
```

Comment:
We also need to delete this setting when the module is connected.

Reply:
Is there a reason to do this? Cause the SetupForm is only ever used when the module does not have a client ID saved. So the existing temporary client ID will never overwrite the current client ID in the form. It does work fine in my testing unless I am missing a use case. If we disconnect the module again with another client ID, the latest client ID then persists.

If there is a valid reason to do this, which is the best callback to add this logic to? Because doing this in `on_activation()` will delete the setting as soon as we reach the `SetupForm` and it doesn't auto-fill the text field anymore. In the backend, we don't have a 'on_connection' callback.  We could create a new endpoint and call this endpoint from `onSetupComplete` - but it will be unnecessary code if there is no purpose here?

Reply:
> Is there a reason to do this?

There is no reason to keep it after we connect the module - this is the reason why we need to delete it. I know it won't hurt keeping it but it is better to wipe out the table after the dinner, not before it.

> ... which is the best callback to add this logic to?
> In the backend, we don't have a 'on_connection' callback.

We can use action hooks to implement that: https://github.com/google/site-kit-wp/blob/bf1bec6e07f7172f4fda5222efc5306e17fccbcd/includes/Core/Modules/REST_Modules_Controller.php#L471-L474

It can be something like this:

```php
add_action( 'googlesitekit_pre_save_settings_; . self::MODULE_SLUG, function() {
    if ( $this->is_connected() ) {
        return;
    }

    add_action( 'googlesitekit_save_settings_'  . self::MODULE_SLUG, function() {
        if ( $this->is_connected() ) {
            // delete the previous client ID.
        }
    } );
} );
```

Reply:
>it is better to wipe out the table after the dinner...

😄 Lol agreed!